### DISCUSSION TITLE
Real-time events in JSON-RPC Relay

### DISCUSSION THREAD

**Discussion by Ivo-Yankov** [hiero-ledger/hiero-improvement-proposals#695] [Edits]
The [HIP](https://github.com/hashgraph/hedera-improvement-proposal/pull/694) proposes adding the `eth_subscribe` and `eth_unsubscribe` methods to the JSON-RPC Relay

**Comment by iron4548** [comment]
I'd be curious if the real-time events will become available outside of JSON-RPC? And if so, is there a HIP or a Git issue I can track? Thanks

**Comment by nickpoorman** [comment]
@iron4548 The network produces a record stream. If you look at the mirror node importer code you'll see how that record stream is fetched, parsed, and inserted into the mirror node database.

Anyone can pull the record stream files from the AWS S3 or GCP bucket themselves.

**Comment by iron4548** [comment]
A question, would be it be possible to omit the address when subscribing to events using an array of topic0 ids only? For example, DEX pair contracts have events that share the same topic0 id signature, and there are more than one pair contracts (in the hundreds, if not thousands).
